admin_guide/registry_data_check_restore.adoc

```diff
@@ -0,0 +1,112 @@
+to understand which imagestreams were damaged.
+
+. Necessary images were removed from the {product-title} cluster's
+database, but remained in the OpenShift Container Registry storage. This
+may occur due to incorrect pruning settings or due to accidental manual
+removal. In this case, the problem is to restore the references in the
+{product-title} cluster's database in order to prevent blobs from being
+deleted during the hard pruning process.
+
+== Preparation
+
+Before start, you must give the registry's service account permissions to
+change the {product-title} cluster's database:
++
+----
+$ service_account=$(oc get -n default \
+    -o jsonpath=$'system:serviceaccount:{.metadata.namespace}:{.spec.template.spec.serviceAccountName}\n' \
+    dc/docker-registry)
+$ oc adm policy add-cluster-role-to-user cluster-admin ${service_account}
+----
+
+[NOTE]
+====
+After completion of the process of restoring the database and storage
+consistency, do not forget to revoke the granted privileges.
+====
+
+== Find broken objects
+
+To list broken imagestream tags and images run (no changes are actually made):
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -restore-mode=check
+----
+
+If the {product-title} cluster's database is too big and you do not want make
+full check then you can limit the scan and check only the database or only the storage:
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -restore-mode=check-database
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -restore-mode=check-storage
+----
+
+You can also limit the namespace you want to scan:
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -restore-mode=check -restore-namespace=myproject
+----
+
+== Restore imagesstreams
+
+To recreate lost imagestream tags based on the state of the storage, use the command:
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -restore-mode=recover
+----
+
+Unfortunately, imagestream tag names are not stored on the storage.
+Therefore, they can not be restored. Also, if the imagestream tag had a
+history, then it also could not be restored. Tag creation time will also
+be lost. All images from the history will be restored as separate tags.
+The recovered tag names will have the form: `lost-found-<DIGEST>`. 
```

Comment:
this tag is created in whatever namespace/imagestream the content was originally associated with?

Reply:
@bparees If there is no imagestream, then we will not create it again and return error. If the imagestream already has such tag, we will also return an error.

Reply:
ok, might be worth describing that behavior here.